lib/segment/src/types.rs

```diff
@@ -233,6 +234,79 @@ impl PartialEq for ScoredPoint {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub enum Score {
+    Int(i64),
+    Float(f64),
+}
+
+impl Eq for Score {}
+
+impl PartialOrd for Score {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Score {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (self, other) {
+            (Score::Float(f), Score::Int(i)) => {
+                OrderingValue::Float(*f).cmp(&OrderingValue::Int(*i))
+            }
+            (Score::Int(i), Score::Float(f)) => {
+                OrderingValue::Int(*i).cmp(&OrderingValue::Float(*f))
+            }
+            (Score::Int(self_i), Score::Int(other_i)) => self_i.cmp(other_i),
+            (Score::Float(self_f), Score::Float(other_f)) => {
+                OrderedFloat(*self_f).cmp(&OrderedFloat(*other_f))
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct QueriedPoint {
+    /// Point id
+    pub id: PointIdType,
+    /// Point version
+    pub version: Option<SeqNumberType>,
+    /// Points vector distance to the query vector
+    pub score: Option<Score>,
+    /// Payload - values assigned to the point
+    pub payload: Option<Payload>,
+    /// Vector of the point
+    pub vector: Option<VectorStruct>,
+    /// Shard Key
+    pub shard_key: Option<ShardKey>,
+}
+
+impl Eq for QueriedPoint {}
+
+impl Ord for QueriedPoint {
+    /// A point with a score is always better than point without a score
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (&self.score, &other.score) {
+            (Some(a), Some(b)) => a.cmp(b),
+            (None, Some(_)) => Ordering::Less,
+            (Some(_), None) => Ordering::Greater,
+            (None, None) => self.id.cmp(&other.id),
+        }
```

Comment:
`Option` implements `Ord` where `None` is always less, so you could do this with a bit less branching:
 
```suggestion
        if self.score.is_some() || other.score.is_some() {
            self.score.cmp(&other.score)
        } else {
            self.id.cmp(&other.id)
        }
```

Or maybe you want this instead:

```rust
self.score.cmp(&other.score).then_with(|| self.id.cmp(&other.id))
```